docs/packages/function_tree.md

```diff
@@ -76,12 +85,36 @@ You can now use the debugger from your functions contexts and/or providers:
 function someFunction(context) {
   context.debugger.send({
     method: 'someMethod',
-    args: ['foo', 'bar'],
-    color: 'red'
+    args: ['foo', 'bar']
   })
 }
 ```
 
+Or you can use it when creating providers to easily wrap their usage:
+
+```js
+function MyProvider (options = {}) {
+  let cachedProvider = null
+
+  function createProvider (context) {
+    return {
+      doSomething() {},
+      doSomethingElse() {}
+    }
+  }
+
+  return (context) => {
+    context.myProvider = cachedProvider = cachedProvider || createProvider(context)
```

Comment:
Hm.... maybe we should just get prettier in there... so this stuff happens automatically